tests/src/Integration/Action/BlockIPTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\BlockIPTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\BlockIP
+ * @group rules_action
+ */
+class BlockIPTest extends RulesIntegrationTestBase {
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * @var \PHPUnit_Framework_MockObject_MockObject|\Drupal\ban\BanIpManagerInterface
+   */
+  protected $banManager;
+
+  /**
+   * @var \PHPUnit_Framework_MockObject_MockObject|\Symfony\Component\HttpFoundation\Request
+   */
+  protected $request;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    // We need the ban module.
+    $this->enableModule('ban');
+    $this->banManager = $this->getMock('Drupal\ban\BanIpManagerInterface');
+    $this->container->set('ban.ip_manager', $this->banManager);
+
+    $this->request = $this->getMock('Symfony\Component\HttpFoundation\Request');
+    $this->container->set('request', $this->request);
+
+    $this->action = $this->actionManager->createInstance('rules_block_ip');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary() {
+    $this->assertEquals('Blocks an IP address', $this->action->summary());
+  }
+
+  /**
+   * Tests the action execution with Context IP.
+   *
+   * @covers ::execute
+   */
+  public function testActionExecutionWithContextIP() {
+    $ip = '192.155.122.111';
```

Comment:
See https://tools.ietf.org/html/rfc3330 : "192.0.2.0/24 - This block is assigned as "TEST-NET" for use in
   documentation and example code.  It is often used in conjunction with
   domain names example.com or example.net in vendor and protocol
   documentation.  Addresses within this block should not appear on the
   public Internet."

So let's use an IP from that range.